packages/design-system/src/components/Dialog/useBodyScrollPrevention.ts

```diff
@@ -0,0 +1,59 @@
+import { Reducer, useReducer, useLayoutEffect } from 'react';
+
+const CLASS_NAME = 'ds--dialog-open';
+const PROPERTY_NAME = '--body_top--dialog-open';
+
+interface OpenState {
+  name: 'open';
+  bodyScrollY: number;
+}
+
+interface ClosedState {
+  name: 'closed';
+}
+
+const OPEN = (_state: ClosedState): OpenState => {
```

Comment:
This `OPEN` action requires a starting point of `ClosedState` and an ending point of `OpenState`.